include/oneapi/dpl/internal/scan_by_segment_impl.h

```diff
@@ -392,9 +390,14 @@ __scan_by_segment_impl_common(__internal::__hetero_tag<_BackendTag>, Policy&& po
 
     constexpr iter_value_t identity = unseq_backend::__known_identity<Operator, iter_value_t>;
 
-    __sycl_scan_by_segment_impl<Inclusive::value>()(_BackendTag{}, ::std::forward<Policy>(policy), key_buf.all_view(),
-                                                    value_buf.all_view(), value_output_buf.all_view(), binary_pred,
-                                                    binary_op, init, identity);
+    using _CustomName = oneapi::dpl::__internal::__policy_kernel_name<Policy>;
+
+    sycl::queue __q_local = policy.queue();
+
+    __sycl_scan_by_segment_impl<_CustomName, Inclusive::value>{}(
+        oneapi::dpl::__internal::__device_backend_tag{}, __q_local, key_buf.all_view(), value_buf.all_view(),
+        value_output_buf.all_view(), binary_pred, binary_op, init, identity);
+
```

Comment:
Assuming we all agree this stays in the `hetero` space, this change should be reverted.

Reply:
We have only one implementation of `__scan_by_segment_impl_common`, `__sycl_scan_by_segment_impl` - and it's for sycl backend.
Also, the `__sycl_scan_by_segment_impl` is exactly submitter, so it shouldn't work with `_ExecutionPolicy`.
So I believe that this change looks correct but these two things just should be moved into `sycl` backend.

Reply:
I think that would be great.

Reply:
I've created this PR #2201

Reply:
I approved you PR https://github.com/uxlfoundation/oneDPL/pull/2201 and merged it into this branch right now.

Reply:
So I resolve this conversation.